core/src/main/scala/better/files/ThreadBackedFileMonitor.scala

```diff
@@ -60,15 +57,12 @@ abstract class ThreadBackedFileMonitor(val root: File, maxDepth: Int) extends Fi
     }
   }
 
-  override def start() = {
+  override def start()(implicit executionContext: ExecutionContext) = {
     watch(root, maxDepth)
-    thread.start()
+    executionContext.execute(() => Iterator.continually(service.take()).foreach(process))
```

Comment:
Is this scheduling a thread (on the pool) to continually call service.take then process (handle the event) on the same thread? As there's only one WatchService, doesn't this mean there'll only be one thread active from the pool?